### PR TITLE
Update test_coast_aliases

### PR DESCRIPTION
**Pull request by willschlitzer**
The current `test_coast_aliases()` in `test_coast.py` does not include all of the current aliases in PyGMT v0.2.1; I updated the function to include all of the aliases and changed the test format to use `@check_figures_equal`.

### CODE REVIEW

**Comment by seisman** [comment]
Following https://github.com/GenericMappingTools/pygmt/issues/771#issuecomment-751854298, I think we don't want the changes in this PR. Am I right?

**Comment by willschlitzer** [comment]
@seisman Sorry if my poor timing of submitting this pull request and #771 at the same time made this confusing. In my opinion, I think the changes in this PR are valid. I'll admit that I'm not too experienced with writing tests, but I think one of the minimum standards for PyGMT tests should be including a test that makes sure all of the aliases are successfully passed to the GMT API. My thought process here is this ensures that any changes in the GMT argument names, or any accidental changes to the PyGMT aliases, are caught. The test already existed, but my assumption is that `Figure.coast` has expanded to include more arguments since the test was written, so I wanted to make sure that tests all of the aliases/parameters that `Figure.coast` accepts.

File: pygmt/tests/test_coast.py [hunk 1]
```diff
@@ -37,23 +37,49 @@ def test_coast_iceland():
...
+        J="M25c",
+        B="afg",
+        I="1/1p,black",
+        N="1/0.5p,-",
+        W="0.25p,white",
+        G="moccasin",
+        S="skyblue",
+        D="i",
+        A=1000,
+        L="jMM+c1+w1000k+f+l",
```
Comment (seisman) [inline pygmt/tests/test_coast.py:55]: `MM` is not a valid anchor. Do you mean `CM`?

Comment (willschlitzer) [reply pygmt/tests/test_coast.py]: @seisman I did mean `CM`, and changed it in my most recent commit. I don't know enough about how GMT handles a different input, but it looks like `MM` is a valid anchor; both argument variations look the same on the figure. I ran a `check_figures_equal` test for the two and it passed.

Comment (seisman) [reply pygmt/tests/test_coast.py]: I guess that GMT may default to "CM" if the given anchor can not be recognized.

File: pygmt/tests/test_coast.py [hunk 2]
```diff
@@ -37,23 +37,49 @@ def test_coast_iceland():
...
+        I="1/1p,black",
+        N="1/0.5p,-",
+        W="0.25p,white",
+        G="moccasin",
+        S="skyblue",
+        D="i",
+        A=1000,
+        L="jCM+c1+w1000k+f+l",
+        U=True,
+        V="q",
```
Comment (seisman) [inline pygmt/tests/test_coast.py:57]:
```suggestion
```

File: pygmt/tests/test_coast.py [hunk 3]
```diff
@@ -37,23 +37,49 @@ def test_coast_iceland():
...
+        rivers="1/1p,black",  # I
+        borders="1/0.5p,-",  # N
+        shorelines="0.25p,white",  # W
+        land="moccasin",  # G
+        water="skyblue",  # S
+        resolution="i",  # D
+        area_thresh=1000,  # A
+        map_scale="jCM+c1+w1000k+f+l",  # L
+        timestamp=True,  # U
+        verbose="q",  # V
```
Comment (seisman) [inline pygmt/tests/test_coast.py:76]:
```suggestion
```

**Review by seisman — COMMENTED** [review]
The test looks OK, but it's a bad idea to use **-Vq** which disables warnings and errors.

File: pygmt/tests/test_coast.py (hunk 3 above)
Comment (willschlitzer) [reply pygmt/tests/test_coast.py:76]: I included that to make sure that all of the aliases are tested; I'm guessing we don't need to test `verbose`?

**Review by seisman — APPROVED** [review]
@willschlitzer The test looks good to me.

As you're now a member of the python-maintainers team, you should be able to merge the  PR when the CI tests finish. Please try it.

It's OK if one or two Windows CI tests fail. That's a known issue (https://github.com/GenericMappingTools/pygmt/pull/758) and we don't have a solution yet .

**Comment by willschlitzer** [comment]
@seisman sounds good!

File: pygmt/tests/test_coast.py [hunk 4]
```diff
@@ -37,23 +37,47 @@ def test_coast_iceland():
...
+        B="afg",
+        I="1/1p,black",
+        N="1/0.5p,-",
+        W="0.25p,white",
+        G="moccasin",
+        S="skyblue",
+        D="i",
+        A=1000,
+        L="jCM+c1+w1000k+f+l",
+        U=True,
```
Comment (seisman) [inline pygmt/tests/test_coast.py:56]: @willschlitzer It's problematic to use `U=True` here, as it creates a timestamp and the timestep can be different for the reference and test images, which can cause failure.

**Comment by seisman** [comment]
@willschlitzer It's problematic to use `U=True` here, as it creates a timestamp and the timestep can be different for the reference and test images, which can cause failure.

**Pull request merged**